app/code/community/Bolt/Boltpay/controllers/OnepageController.php

```diff
@@ -92,4 +92,29 @@ public function saveShippingMethodAction()
             throw $e;
         }
     }
+
+    /**
+     * Order success action
+     */
+    public function successAction()
```

Comment:
Unfortunately, we are overriding the same method here and there is no clear way to distinguish if this is a product page order or a normal order.  

https://github.com/BoltApp/bolt-magento1/blob/cadf0069d45a6570253310aa39a9c40a3f717cf9/app/code/community/Bolt/Boltpay/controllers/OnepageController.php#L43-L80

Since you wish to pursue this without adding something to the DB then my suggestion is to pass a URL GET paramater for `is_bolt_pdp`.  However, by far, I think the far better idea is to extend the order model table with this `is_bolt_pdp`.  With this flag, you will not need to override the success URL at all and we will always be able to clearly tell which orders came from product page and which from some other source.  Is there a reason why you are not wanting to extend the order model?